### DISCUSSION TITLE
Creation of Instance with List

### DISCUSSION THREAD

**Discussion by bspautz1969** [dadhi/DryIoc#389] [Q&A]
Good Day!

I have a dependency in a created instance of a class. This dependency is of the type IList <Class>.

It is going to created correctly. 

However, I cannot add anything to this list in the instance created.

I always get the error "Collection was of a fixed size."

I got it through the normal constructor creation as well as through the Register_with_instance_factory_method.

Do you have a tip for me?

**Comment by dadhi** [comment]
Wrap the injected list in a new one then add anything to it.

**Reply by dadhi** [reply]
But the first question whould be why do you need this?

**Reply by bspautz1969** [reply]
The injected List is a dependency on my class. And i must be able to add items to it. And that´s not possible right now.

**Reply by dadhi** [reply]
What will you add to it?

Ok, nevermind. Here is the example with the wrapping in the new list:

```cs
class A {
  List<X> _xs;
  public A(IList<X> xs) {
     _xs = new List<X>(xs);
  }
}
```

**Reply by bspautz1969** [reply]
Sometimes you just stand in your own way and can't see the forest for the trees.

Thanks man, it works ....
